### PR TITLE
Bug fix in rational.cpp

### PR DESCRIPTION
**Pull request by nishnik**
@isuruf 
Should the whole if statement be removed ?

### CODE REVIEW

**Comment by isuruf** [comment]
Yes, den can never be zero

File: symengine/rational.cpp [hunk 1]
```diff
@@ -128,6 +125,9 @@ bool Rational::is_perfect_power(bool is_expected) const
 
 bool Rational::nth_root(const Ptr<RCP<const Number>> &the_rat, unsigned long n) const
 {
+    if (n == 0)
+        throw std::runtime_error("i_nth_root: Can not find Zeroth root");
+
```
Comment (nishnik) [inline symengine/rational.cpp]: @isuruf 
I tried running a test with `n` set to `0`.
I got:
```
due to a fatal error condition:
  SIGFPE - Floating point error signal
  Traceback (most recent call last):
    File unknown, in _start()
    File "/build/buildd/glibc-2.21/csu/libc-start.c", line 289, in
  __libc_start_main()
    File "/home/nishnik/git_symengine/symengine/symengine/utilities/catch/catch
  .cpp", line 19, in main()
      int result = Catch::Session().run( argc, argv );
    File "/home/nishnik/git_symengine/symengine/symengine/utilities/catch/catch
  .hpp", line 5662, in Catch::Session::run(int, char* const*)
      returnCode = run();
    File "/home/nishnik/git_symengine/symengine/symengine/utilities/catch/catch
  .hpp", line 5682, in Catch::Session::run()
      return static_cast<int>( runner.runTests().assertions.failed );
    File "/home/nishnik/git_symengine/symengine/symengine/utilities/catch/catch
  .hpp", line 5551, in Catch::Runner::runTests()
      totals += context.runTest( *it );
    File "/home/nishnik/git_symengine/symengine/symengine/utilities/catch/catch
  .hpp", line 5230, in Catch::RunContext::runTest(Catch::TestCase const&)
      runCurrentTest( redirectedCout, redirectedCerr );
    File "/home/nishnik/git_symengine/symengine/symengine/utilities/catch/catch
  .hpp", line 5393, in Catch::RunContext::runCurrentTest(std::string&,
  std::string&)
      invokeActiveTestCase();
    File "/home/nishnik/git_symengine/symengine/symengine/utilities/catch/catch
  .hpp", line 5421, in Catch::RunContext::invokeActiveTestCase()
      m_activeTestCase->invoke();
    File "/home/nishnik/git_symengine/symengine/symengine/utilities/catch/catch
  .hpp", line 6727, in Catch::TestCase::invoke() const
      test->invoke();
    File "/home/nishnik/git_symengine/symengine/symengine/utilities/catch/catch
  .hpp", line 5822, in Catch::FreeFunctionTestCase::invoke() const
      m_fun();
    File "/home/nishnik/git_symengine/symengine/symengine/tests/basic
  /test_rational.cpp", line 68, in ____C_A_T_C_H____T_E_S_T____59()
      std::cout<<q3_5->nth_root(outArg(res),0);
    File "/home/nishnik/git_symengine/symengine/symengine/rational.cpp", line
  132, in SymEngine::Rational::nth_root
  (Teuchos::Ptr<Teuchos::RCP<SymEngine::Number const> > const&, unsigned long)
  const
      int ret = mpz_root(r.get_num_mpz_t(), i.get_num_mpz_t(), n);
    File unknown, in __gmpz_root()
    File unknown, in __gmp_divide_by_zero()
    File unknown, in __gmp_exception()
    File unknown, in killpg()
```
So, I think this exception must be thrown.

Comment (nishnik) [inline symengine/rational.cpp]: This exception throwing is also in integer.cpp

Comment (nishnik) [inline symengine/rational.cpp]: Ping @isuruf

Comment (isuruf) [inline symengine/rational.cpp]: Can you add a test which checks that an error is thrown in n=0 case?

Comment (nishnik) [inline symengine/rational.cpp]: Done.

**Comment by isuruf** [comment]
Thanks for the PR

**Pull request merged**